jbt-it/neomdb
# 457 server trainee application admin page

How to test:
- test route post /generation  to create a new generation expects type NewGenerationRequestDto
  - requestBody = {
        description: "Wintersemester 24/25",
        applicationStart: new Date("2024-10-01T22:00:00.000Z"),
        applicationEnd: new Date("2024-10-17T22:00:00.000Z"),
        selectionWeDateStart: new Date("2024-10-18T22:00:00.000Z"),
        selectionWeDateEnd: new Date("2024-10-20T22:00:00.000Z"),
        wwDateStart: new Date("2024-11-01T22:00:00.000Z"),
        wwDateEnd: new Date("2024-11-03T22:00:00.000Z"),
      }

- test route patch /generation to update an existing generaiton expects type GenerationDto
  - requestBody = {
        generationId: 15,
        description: new Date("2024-10-01T22:00:00.000Z"),
        applicationStart: new Date("2024-10-01T22:00:00.000Z"),
        applicationEnd: new Date("2024-10-17T22:00:00.000Z"),
        selectionWeDateStart: new Date("2024-10-18T22:00:00.000Z"),
        selectionWeDateEnd: new Date("2024-10-20T22:00:00.000Z"),
        wwDateStart: new Date("2024-11-01T22:00:00.000Z"),
        wwDateEnd: new Date("2024-11-03T22:00:00.000Z"),
        infoEveningVisitors: 100,
        doorCode: "12345",
        electionStart: new Date("2024-11-04T22:00:00.000Z"),
        electionEnd: new Date("2024-11-11T22:00:00.000Z"),
      }

- test route get /evaluations/{id} to get the evaluations for the trainees for a member (EV)

- test route get /feedback to get the feedback statitics

- test route post /invitation to set the invited indicator
eg. { applicantIds: [123,456] } (you need to check the applicantIds in your own db)

- test route post /admission to set the admission indicator and create a new member
eg. { applicantIds: [123,456] } (you need to check the applicantIds in your own db)
